concierge/service/src/main/java/org/eclipse/ditto/concierge/service/enforcement/DefaultCreationRestrictionEnforcer.java

```diff
@@ -0,0 +1,129 @@
+ */
+@Immutable
+public class DefaultCreationRestrictionEnforcer implements CreationRestrictionEnforcer {
+
+    private static final Logger log = DittoLoggerFactory.getThreadSafeLogger(DefaultCreationRestrictionEnforcer.class);
+
+    private final EntityCreationConfig config;
+
+    private DefaultCreationRestrictionEnforcer(final EntityCreationConfig config) {
+        this.config = config;
+    }
+
+    @Override
+    public boolean canCreate(final Context context) {
+        return matchesList(this.config.getGrant(), context)
+                && !matchesList(this.config.getRevoke(), context);
+    }
+
+    private boolean matchesList(final List<CreationRestrictionConfig> list, final Context context) {
+        return list.stream().anyMatch(entry -> matches(entry, context));
+    }
+
+    private static boolean matches(final CreationRestrictionConfig entry, final Context context) {
+
+        return matchesResourceType(entry.getResourceTypes(), context)
+                && matchesAuthSubject(entry.getAuthSubjects(), context)
+                && matchesNamespacePattern(entry.getNamespaces(), context);
+
+    }
+
+    private static boolean matchesResourceType(final Set<String> resourceTypes, final Context context) {
+        if (resourceTypes.isEmpty()) {
+            // no restriction -> pass
+            log.debug("No resource type restriction: pass");
+            return true;
+        }
+
+        if ( resourceTypes.contains(context.getResourceType())) {
+            return true;
+        }
+
+        log.debug("No resource type match: reject");
+        // no match, but non-empty list -> reject
+        return false;
+    }
+
+    private static boolean matchesAuthSubject(final Set<String> authSubjects, final Context context) {
+        if (authSubjects.isEmpty()) {
+            // no restrictions -> pass
+            log.debug("No auth subject restriction: pass");
+            return true;
+        }
+
+        for (final var subject : context.getAuthorizationContext().getAuthorizationSubjectIds()) {
+            if (authSubjects.contains(subject)) {
+                log.debug("Matched auth subject {}: pass", subject);
+                // entry found -> pass
+                return true;
+            }
+        }
+
+        log.debug("No auth subject match: reject");
+        // no match, but non-empty list -> reject
+        return false;
+    }
+
+    private static boolean matchesNamespacePattern(final List<Pattern> namespacePatterns, final Context context) {
+
+        if (namespacePatterns.isEmpty()) {
+            log.debug("No namespace restriction: pass");
+            // no restrictions -> pass
+            return true;
+        }
+
+        var namespace = context.getNamespace();
+        for (final var namespacePattern : namespacePatterns) {
+            if (namespacePattern.matcher(namespace).matches()) {
+                log.debug("Namespace '{}' matched {}: pass", namespace, namespacePattern);
```

Comment:
please add correlationId / DittoHeaders extracted from `context` to log.
In order to not do this for each "for" iteration the new logger can also be created once in the surrounding method.